src/Catalyst.Cli/Commands/AddFileCommand.cs

```diff
@@ -0,0 +1,114 @@
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System.IO;
+using System.Threading;
+using Catalyst.Abstractions.Cli.Commands;
+using Catalyst.Abstractions.FileTransfer;
+using Catalyst.Abstractions.Types;
+using Catalyst.Cli.CommandTypes;
+using Catalyst.Cli.Options;
+using Catalyst.Core.Lib.Extensions;
+using Catalyst.Core.Lib.FileTransfer;
+using Catalyst.Core.Lib.IO.Messaging.Dto;
+using Catalyst.Protocol.Rpc.Node;
+using Serilog;
+
+namespace Catalyst.Cli.Commands
+{
+    public sealed class AddFileCommand : BaseMessageCommand<AddFileToDfsRequest, AddFileToDfsResponse, AddFileOptions>
+    {
+        public static string ErrorNoResponseCodes => "No response codes were found.";
+        private readonly IUploadFileTransferFactory _uploadFileTransferFactory;
+
+        public AddFileCommand(IUploadFileTransferFactory uploadFileTransferFactory,
+            ICommandContext commandContext,
+            ILogger logger) : base(commandContext, logger)
+        {
+            _uploadFileTransferFactory = uploadFileTransferFactory;
+        }
+
+        protected override AddFileToDfsRequest GetMessage(AddFileOptions option)
+        {
+            return new AddFileToDfsRequest
+            {
+                FileName = Path.GetFileName(option.File)
+            };
+        }
+
+        protected override bool ExecuteCommand(AddFileOptions options) { return File.Exists(options.File); }
+
+        public override void SendMessage(AddFileOptions options)
+        {
+            var request = GetMessage(options);
+
+            using (var fileStream = File.Open(options.File, FileMode.Open))
+            {
+                request.FileSize = (ulong) fileStream.Length;
+            }
+
+            var protocolMessage = request.ToProtocolMessage(SenderPeerId);
+            var requestMessage = new MessageDto(
+                protocolMessage,
+                RecipientPeerId
+            );
+
+            IUploadFileInformation fileTransfer = new UploadFileTransferInformation(
+                File.Open(options.File, FileMode.Open),
+                SenderPeerId,
+                RecipientPeerId,
+                Target.Channel,
+                requestMessage.CorrelationId);
+
+            _uploadFileTransferFactory.RegisterTransfer(fileTransfer);
+
+            Target.SendMessage(requestMessage);
+
+            while (!fileTransfer.ChunkIndicatorsTrue() && !fileTransfer.IsExpired())
+            {
+                CommandContext.UserOutput.Write($"\rUploaded: {fileTransfer.GetPercentage().ToString()}%");
+                Thread.Sleep(500);
+            }
+
+            if (fileTransfer.ChunkIndicatorsTrue())
+                CommandContext.UserOutput.Write($"\rUploaded: {fileTransfer.GetPercentage().ToString()}%\n");
+            else
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'else' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4594394)